core/src/main/java/org/pentaho/platform/scheduler2/action/DefaultActionInvoker.java

```diff
@@ -132,8 +133,12 @@ protected IActionInvokeStatus invokeActionImpl( final IAction actionBean,
     ActionUtil.removeKeyFromMap( params, ActionUtil.INVOKER_STREAMPROVIDER );
     ActionUtil.removeKeyFromMap( params, ActionUtil.INVOKER_UIPASSPARAM );
 
-    final ActionRunner
-      actionBeanRunner = new ActionRunner( actionBean, actionUser, params, streamProvider );
+    final IActionRunner actionBeanRunner = PentahoSystem.get( IActionRunner.class );
```

Comment:
allowing for enterprise version of class in upstream PR `EnterpriseActionRunner`